docs/assignment2.md

```diff
@@ -1 +1,233 @@
+
+#### **Player**
+
+Class representing a single player. Only exists within a **Game** and is created from user input.
+
+Attributes 
+
+- *name* (type: String): player's name
+- *score* (type: int): player's score
+
+Operations
+
+- *String getName()*: returns *name*
+- *int getScore()* : returns *score*
+
+Associations
+
+All **Player** objects are contained within **Game**. A team is an aggregation of exactly 8 **Player** objects. 
+
+
+
+#### **Team**
+
+Class representing a single soccer team. Only exists within a **Game** and is created through the GUI by the user through the selection of 8 **Player** objects.
+
+Attributes 
+
+- *players* (type: Player[]): all the players in a team. Represented by an array as a team has exactly 8 players.
+- *score* (type: int): team's total score
+- *wins* (type: int): team's total number of wins in leagues played by the user; initilized to 0
+- *teamName* (type: String): team name, unique attribute used to identify a team
+
+Operations 
+
+- *int calculateScore()*: calculates the *score* of a team by adding up all the *players* objects individual scores
+-  *int getTeamScore()*: returns *score*
+-  *int getTotalWins()*: returns *wins*
+- *String getTeamName()*: returns *teamName*
+- *void addNewWin()*: increments *wins*
+
+Associations
+
+The composition relation between **Team** and **Game** suggests the fact that all **Team** objects ever created are part of the **Game**.allTeams. 
+Each team is an aggregation of exactly 8 **Player** objects.  
+
+#### **League** 
+
+A league simulates multiple rounds of matches between user-selected teams and determines the winners.
+
+Attributes
+
+- *initialTeams* (type:Queue<Team&gt;): List of the teams playing in the first round, as selected by the user.  Already described in **Game**.createLeague's parameter as: list of teams ordered by their playing order, e.g.  a round with matches (Team1 vs. Team2) and (Team3 vs. Team4) would be represented as {Team1, Team2, Team3, Team4}
+-  *winners* (type: HashMap<Integer, ArrayList<Team&gt;>):  A key represents the round number, e.g. first round is 1, while the values represent the winners at each given round.
+
+Operations
+
+- *void playRound(Queue<Team&gt; teamsToPlay, int round)*: recursive method that generates *winners*. First call is *playRound(initialTeams, 1)*. The method calls itself for following rounds until one team remains (the league winner) 
+- HashMap<Integer, ArrayList<Team>> getWinners(): calls *playRound* method and returns *winners*
+
+Associations
+
+A single **League** object can exist at one time within a **Game**. A league can start with at most 16 teams as per requirement **FR6**. Within the *playRound* method, **Match** objects are created from the playing teams to generate a winner.
```

Comment:
Since you mentioned in the feature description that the first round can have at most 16 matches, it only makes sense that this association would have an end multiplicity of 0..16, since can logically not spawn more than 16 match objects at one time